readthedocs/rtd_tests/tests/test_version_slug.py

```diff
@@ -0,0 +1,46 @@
+from django.test import TestCase
+
+from builds.models import Version
+from builds.version_slug import VersionSlugField
+from projects.models import Project
+
+
+class VersionSlugFieldTests(TestCase):
+    fixtures = ["eric", "test_data"]
+
+    def setUp(self):
+        self.pip = Project.objects.get(slug='pip')
+
+    def test_saving(self):
+        version = Version.objects.create(
+            verbose_name='1.0',
+            project=self.pip)
+        self.assertEqual(version.slug, '1.0')
+
+    def test_normalizing(self):
+        version = Version.objects.create(
+            verbose_name='1%0',
+            project=self.pip)
+        self.assertEqual(version.slug, '1-0')
+
+    def test_normalizing_slashes(self):
+        version = Version.objects.create(
+            verbose_name='releases/1.0',
+            project=self.pip)
+        self.assertEqual(version.slug, 'releases__1.0')
```

Comment:
Still not sure about making ugly slugs by default here. I think we could pretty safely creates ``releases-1.0`` here, which is much nicer to read. I'm guessing we don't have a huge number of slugs that will have collisions between "/" and "-" in their names in the same repo.

We could do a query on prod, but my guess is there are 0 matching branch names that collide.